src/typeguard/_transformer.py

```diff
@@ -429,3 +478,105 @@ def visit_Yield(self, node: Yield):
+                    names = [target]
+                elif isinstance(target, Tuple):
+                    names = target.elts
+                else:
+                    continue
+
+                for name in names:
+                    annotation = self._memo.variable_annotations.get(name.id)
+                    if annotation is not None:
+                        annotations_[name.id] = annotation
+
+            if annotations_:
+                func_name = self._get_typeguard_import("check_variable_assignment")
+                keys = [Constant(argname) for argname in annotations_.keys()]
+                values = list(annotations_.values())
+                expected_types = Dict(keys=keys, values=values)
+                node.value = Call(
+                    func_name,
+                    [node.value, expected_types, self._memo.get_call_memo_name()],
+                    [],
+                )
+
+        return node
+
+    def visit_NamedExpr(self, node: NamedExpr) -> Any:
+        # Only instrument function-local assignments
+        if isinstance(self._memo.node, (FunctionDef, AsyncFunctionDef)) and isinstance(
+            node.target, Name
+        ):
+            # Bail out if no matching annotation is found
+            annotation = self._memo.variable_annotations.get(node.target.id)
+            if annotation is None:
+                return node
+
+            func_name = self._get_typeguard_import("check_variable_assignment")
+            expected_types = Dict(keys=[Constant(node.target.id)], values=[annotation])
+            node.value = Call(
+                func_name,
+                [node.value, expected_types, self._memo.get_call_memo_name()],
+                [],
+            )
+
+        return node
+
+    def visit_AugAssign(self, node: AugAssign) -> Any:
+        # Only instrument function-local assignments
+        if isinstance(self._memo.node, (FunctionDef, AsyncFunctionDef)) and isinstance(
+            node.target, Name
+        ):
+            # Bail out if no matching annotation is found
+            annotation = self._memo.variable_annotations.get(node.target.id)
+            if annotation is None:
+                return node
+
+            # Bail out if the operator is not found (newer Python version?)
+            try:
+                operator_func = aug_assign_functions[node.op.__class__]
+            except KeyError:
+                return node
+
+            operator_call = Call(
+                Attribute(node.target, operator_func, ctx=Load()), [node.value], []
+            )
+            expected_types = Dict(keys=[Constant(node.target.id)], values=[annotation])
+            check_call = Call(
+                self._get_typeguard_import("check_variable_assignment"),
+                [operator_call, expected_types, self._memo.get_call_memo_name()],
+                [],
+            )
+            node = Assign(targets=[node.target], value=check_call)
+
+        return node
```

Comment:
Looks good, but I would like docstrings to explain the changes on visit for readability, perhaps showing what transformation is made with syntax

Reply:
I've added some docstrings to methods that make changes to the nodes passing through them. Is this satisfactory or would you say it needs more?

Reply:
Looks good! I'm certainly satisfied.